ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/service/support/ArticleParser.java

```diff
@@ -0,0 +1,233 @@
+
+    private List<Article> articleList = new ArrayList<>();
+
+    private static final String CREATED_TAG = "jcr:created";
+    private static final String CONTENT_TAG = "jcr:content";
+    private static final String LAST_MODIFIED_TAG = "cq:lastModified";
+    private static final String UUID_TAG = "jcr:uuid";
+    private static final String TITLE_TAG = "jcr:title";
+    private static final String ROOT_TAG = "root";
+    private static final String FILE_TAG = "fileReference";
+    private static final String BASE_URL = "https://stage.cru.org";
+
+    /**
+     * Constructor
+     * @param jsonObject = the Object to be parsed.
+     */
+    public ArticleParser(JSONObject jsonObject) {
+        this.articleJSON = jsonObject;
+    }
+
+    //region Public Executor
+    /**
+     * This execute function allows you to change the initial jsonObject and run the execution.
+     *
+     * use <code>ARTICLE_LIST_KEY</code> and <code>ATTACHMENT_LIST_KEY</code> to access the Collections
+     * out of the HashMaps
+     *
+     * @param jsonObject = the object to be parsed.
+     * @return = return a list of <code>Article</code> and <code>Attachments</code> in HashMap
+     */
+    public HashMap<String, Object> execute(JSONObject jsonObject) {
+        this.articleJSON = jsonObject;
+        return execute();
+    }
+
+    /**
+     * This executes the parsing of the local JsonObject.
+     *
+     * use <code>ARTICLE_LIST_KEY</code> and <code>ATTACHMENT_LIST_KEY</code> to access the Collections
+     * out of the HashMaps
+     *
+     * @return = return a list of <code>Article</code> and <code>Attachments</code> in HashMap
+     */
+    public HashMap<String, Object> execute() {
+
+        jsonObjectHandler(articleJSON);
+
+        HashMap<String, Object> returnObject = new HashMap<>();
+        returnObject.put(ARTICLE_LIST_KEY, articleList);
+        returnObject.put(ATTACHMENT_LIST_KEY, attachmentList);
+
+        return returnObject;
+    }
+    //endregion public Executor
+
+    //region Article Parsing
+
+    /**
+     * This method takes in a Json Object and Iterates through the keys to determine if they
+     * are Article Objects or Category Object.  Article Objects will be passed on to be parsed
+     * and Category Object will be Recursively placed back into this method.
+     * @param evaluatingObject = the Json Object to be evaluated.
+     */
+    private void jsonObjectHandler(JSONObject evaluatingObject) {
+        // Create loop through Keys
+        Iterator<String> keys = evaluatingObject.keys();
+        while (keys.hasNext()) {
+            String nextKey = keys.next();
+            if (isArticleOrCategory(nextKey)) {
+                try {
+                    JSONObject returnedObject = evaluatingObject.getJSONObject(nextKey);
+                    if (isObjectCategory(returnedObject)) {
+                        jsonObjectHandler(returnedObject);
+                    } else {
```

Comment:
I don't know for sure, but I would assume there might be other types of "objects" that may be present in the json. it's probably safer to use something like:
```java
switch(returnedObject.optString("jcr:primaryType", "")) {
  case "sling:OrderedFolder":
    jsonObjectHandler(returnedObject);
    break;
  case "{{page_type}}":
    parseArticleObject(returnedObject);
    break;
}
```

Reply:
also, `jcr:primaryType`, `sling:OrderedFolder`, and whatever the page type is should be static private constants defined at the top of this class